keps/sig-node/4381-dra-structured-parameters/README.md

```diff
@@ -1462,225 +1278,309 @@ prevented. Deleting the entire cluster also leaves resources allocated outside
+    // to ensure that they remain available while the claim is allocated.
+    //
+    // Entries are list in the same order as in class.config and claim.config,
+    // with class.config entries first.
+    //
+    // +optional
+    Config []DeviceConfiguration
 
-    // VendorClaimParameters are the per-claim configuration parameters
-    // from the resource claim parameters at the time that the claim was
-    // allocated.
-    VendorClaimParameters runtime.Object
+    // RequestName identifies the request in the claim which caused this
+    // device to be allocated. If a request was satisfied by multiple
+    // devices, they are indexed starting with zero in the order
+    // in which they are listed in the results.
+    RequestName string
 
-    // NodeName is the name of the node providing the necessary resources
-    // if the resources are local to a node.
-    NodeName string
+    // This name together with the driver name and the device name field
+    // identify which device was allocated.
+    //
+    // For a driver with node-local devices, it is the name of the node and
+    // the overall ID is `<driver name>/<node name>/<device name>`.
+    //
+    // For other drivers, the overall ID is `<driver name>[/pool
+    // name>]/<device name>`.
+    //
+    // Must not be longer than 253 characters and may contain one or more
+    // DNS sub-domains separated by slashes.
+    //
+    // +optional
+    PoolName string
 
-    // Results lists all allocated driver resources.
-    Results []DriverAllocationResult
+    // DeviceName references one device instance via its name in the driver's
+    // resource pool. It must be a DNS label.
+    DeviceName string
 }
 
-// DriverAllocationResult contains vendor parameters and the allocation result for
-// one request.
-type DriverAllocationResult struct {
-    // VendorRequestParameters are the per-request configuration parameters
-    // from the time that the claim was allocated.
-    VendorRequestParameters runtime.Object
+const PoolNameMaxLength = validation.DNS1123SubdomainMaxLength // Same as for a single node name.
+
+// DeviceConfiguration is one entry in a list of configuration pieces for a device.
+type DeviceConfiguration struct {
+    // Admins is true if the source of the piece was a class and thus
+    // not something that a normal user would have been able to set.
+    Admin bool
 
-    AllocationResultModel
+    DriverConfigurationParameters
 }
 
-// AllocationResultModel must have one and only one field set.
-type AllocationResultModel struct {
-    // NamedResources describes the allocation result when using the named resources model.
-    NamedResources *NamedResourcesAllocationResult
+// DriverConfigurationParameters must have one and only one one field set.
+//
+// In contrast to ConfigurationParameters, the driver name is
+// not included and has to be infered from the context.
+type DriverConfigurationParameters struct {
+    Opaque *runtime.RawExtension
 }
```

Comment:
I think we may be able to drop this, given my comment here:
https://github.com/kubernetes/enhancements/pull/4709/files#r1633116525

Reply:
... still discussing it :grin:

Reply:
Changed - see https://github.com/kubernetes/enhancements/pull/4709#discussion_r1635970009.